extends Control

@onready var pausemenu = $"."
@onready var volumemenu = $Volume
@onready var inputTypeMenu = $InputType

func _process(delta):
	pass
	
	if Input.is_action_pressed("ui_cancel"):
		pausemenu.visible = true
		get_tree().paused = true

func _on_resume_pressed():
	pausemenu.visible = false
	get_tree().paused = false

func _on_back_to_main_menu_pressed():
	pausemenu.visible = false
	get_tree().paused = false
	get_tree().change_scene_to_file("res://Menu/menu.tscn")

func _on_quit_pressed():
	get_tree().quit()

func _on_volume_pressed():
	$CenterContainer.visible = false
	$MarginContainer.visible = false
	volumemenu.visible = true
	
func _on_input_type_button_pressed():
	$CenterContainer.visible = false
	$MarginContainer.visible = false
	inputTypeMenu.visible = true

func _on_back_pressed():
	volumemenu.visible = false
	inputTypeMenu.visible = false
	$CenterContainer.visible = true
	$MarginContainer.visible = true
	pausemenu.visible = true

func _on_input_type_button_item_selected(index):
	if index != -1:
		World.INPUT_SCHEME = index
		EventBus.input_scheme_changed.emit(index)
